### DISCUSSION TITLE
Question about binstubs

### DISCUSSION THREAD

**Discussion by stdedos** [ruby/rubygems#7509] [Q&A]
I have a weird binstub in my repo:

```sh
#!/usr/bin/env sh
"exec" "`dirname $0`/jruby" "$(cd $(dirname $0) && pwd -P)/$(basename $0)" "$@"
# This file was generated by Bundler.
#
# The application 'pry' is installed as part of a gem, and
# this file is here to facilitate running it.
#

require 'pathname'
ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
  Pathname.new(__FILE__).realpath)

require 'rubygems'
require 'bundler/setup'

load Gem.bin_path('pry', 'pry')
```

I regenerated it with `bin/bundle binstubs pry --force --standalone`, but then "expectedly", it leads to `/usr/bin/env: ‘jruby’: No such file or directory` (I understand that the `exec` line is there, because `jruby` is not on my path).
For the record, regenerated (both with and without `--standalone`) it gives a "different" result

However, it leads to `bin/pr:5: warning: unused literal ignored`, which I detest.

- Is the shown bundler format "known" to someone?
- Has anyone had a similar issue before?

**Comment by deivid-rodriguez** [comment]
Can you share the `bin/bundle` binstub you used to generate it? Under which Ruby version did you generate the binstub?

I've seen that kind of "prelude" (first two lines) in the past in scripts generated when compiling Ruby with the `--enable-load-relative` option.

**Reply by stdedos** [reply]
It's actually JRuby (idk if you noticed 😅)

Sure, here's "most" of it:

```sh
#!/usr/bin/env sh
"exec" "`dirname $0`/jruby" "$(cd $(dirname $0) && pwd -P)/$(basename $0)" "$@"

# Exit cleanly from an early interrupt
Signal.trap("INT") { exit 1 }

require 'bundler'
# require_relative '../...'
# require_relative '../...'

# Check if an older version of bundler is installed
# ...

Bundler.settings.temporary(frozen: false)
fail("Could not reset frozen bundle state: #{Bundler.settings.inspect}") if Bundler.frozen_bundle?

# ...
ENV['JARS_SKIP'] ||= 'true'

args = ARGV.dup

if args.empty? || args == %w(install)
    args = %w(install --local)
elsif args == %w(package)
    args = %w(package --all --all-platforms)
end

require 'bundler/friendly_errors'
Bundler.with_friendly_errors do
  require 'bundler/cli'
  Bundler::CLI.start(args, :debug => true)
end
```

**Reply by deivid-rodriguez** [reply]
Unfortunately I have no idea where that binstub comes from, and what's installing the prelude that's causing trouble.

**Reply by stdedos** [reply]
Thank you for trying 😁
